morpheus/stages/general/nonlinear_modules_stage.py

```diff
@@ -0,0 +1,119 @@
+# Copyright (c) 2023, NVIDIA CORPORATION.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+import logging
+import typing
+
+import mrc
+
+from morpheus.config import Config
+from morpheus.pipeline.stage import Stage
+from morpheus.pipeline.stream_pair import StreamPair
+from morpheus.utils.module_utils import load_module
+
+logger = logging.getLogger(__name__)
+
+
+class NonLinearModulesStage(Stage):
```

Comment:
This name isnt great. This class really wraps a multi-port MRC module in a stage and hooks up the ports. Somethink like `MultiPortModuleStage` would be better.